### PR TITLE
TST: Add missing tests for loc slicing of PeriodIndex, TimedeltaIndex

### PR DESCRIPTION
**Pull request by ghost**
While working on the tests for #27060, I found there's no test coverage for `loc` slicing of PeriodIndex and TimedeltaIndex. Added test for string and label slicing.

Someday, it may be a good idea to refactor and parameterize by Index type. But not today.

### CODE REVIEW

File: pandas/tests/indexing/test_loc.py [hunk 1]
```diff
@@ -773,3 +773,35 @@ def test_series_indexing_zerodim_np_array(self):
...
+
+    def test_PeriodIndex_slicing(self):
+        ix = pd.period_range(start='2017-01-01', end='2018-01-01', freq='M')
+        ser = ix.to_series()
+        result = ser.loc[:ix[-2]]
+        expected = ser.iloc[:-1]
+
+        tm.assert_series_equal(result, expected)
+
+    def test_TimedeltaIndex_slicing(self):
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py]: these go in pandas/tests/indexing/test_timedelta.py where some existing tests are

File: pandas/tests/indexing/test_loc.py [hunk 2]
```diff
@@ -773,3 +773,35 @@ def test_series_indexing_zerodim_np_array(self):
...
+
+    def test_TimedeltaIndex_str_slicing(self):
+        ix = pd.timedelta_range(start='1 day', end='2 days', freq='1H')
+        ser = ix.to_series()
+        result = ser.loc[:"1 days"]
+        expected = ser.iloc[:-1]
+
+        tm.assert_series_equal(result, expected)
+
+    def test_PeriodIndex_slicing(self):
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py]: these could go in pandas/tests/indexing/test_datetimes.py or if you want to create a test_period.py would be ok too.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/test_loc.py (hunk 1 above)
Comment (pilkibun) [reply pandas/tests/indexing/test_loc.py]: Done.

File: pandas/tests/indexing/test_loc.py (hunk 2 above)
Comment (pilkibun) [reply pandas/tests/indexing/test_loc.py]: Done.

**Review by jreback — APPROVED** [review]

**Pull request merged**